clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/provider/agent/KubernetesServerGroupCachingAgent.groovy

```diff
@@ -336,7 +336,7 @@ class KubernetesServerGroupCachingAgent extends KubernetesCachingAgent implement
           relationships[Keys.Namespace.LOAD_BALANCERS.ns].addAll(loadBalancerKeys)
         }
 
-        pods.forEach { pod ->
+        if (pods != null) pods.forEach { pod ->
```

Comment:
Would be more idiomatic to instead switch from the java 8 `.forEach` construct to groovy's `.each` construct. (Not sure if there is a reason this wasn't already using `.each`...) You'll want to verify this, but I believe it will not throw NPE's even if applied to a `null` reference.